libs/application/templates/parental-leave/src/fields/Rights/GiveDaysSlider.tsx

```diff
@@ -1,40 +1,61 @@
 import React, { FC, useState } from 'react'
-import { FieldBaseProps, getValueViaPath } from '@island.is/application/core'
-import { Controller, useFormContext } from 'react-hook-form'
-import { Box, Text } from '@island.is/island-ui/core'
+import { FieldBaseProps } from '@island.is/application/core'
+import { useFormContext } from 'react-hook-form'
+import { Box } from '@island.is/island-ui/core'
 import { theme } from '@island.is/island-ui/theme'
 import { useLocale } from '@island.is/localization'
 import { parentalLeaveFormMessages } from '../../lib/messages'
 import Slider from '../components/Slider'
 import BoxChart, { BoxChartKey } from '../components/BoxChart'
-import { maxDaysToGiveOrReceive, defaultMonths, minMonths } from '../../config'
+import { getApplicationAnswers } from '../../lib/parentalLeaveUtils'
+import {
+  maxDaysToGiveOrReceive,
+  defaultMonths,
+  daysInMonth,
+} from '../../config'
+import { YES } from '../../constants'
 
 const GiveDaysSlider: FC<FieldBaseProps> = ({ field, application }) => {
   const { id } = field
   const { formatMessage } = useLocale()
-  const currentAnswer = getValueViaPath(
-    application.answers,
-    field.id,
-    1,
-  ) as number
+  const { register } = useFormContext()
+  const { giveDays } = getApplicationAnswers(application.answers)
+
+  const [chosenGiveDays, setChosenGiveDays] = useState<number>(
+    giveDays === 0 ? 1 : giveDays,
+  )
 
-  const { clearErrors } = useFormContext()
-  const [chosenGiveDays, setChosenGiveDays] = useState<number>(currentAnswer)
   const daysStringKey =
     chosenGiveDays > 1
       ? parentalLeaveFormMessages.shared.giveRightsDays
       : parentalLeaveFormMessages.shared.giveRightsDay
+
   const yourRightsWithGivenDaysStringKey =
     maxDaysToGiveOrReceive - chosenGiveDays === 1
       ? parentalLeaveFormMessages.shared.yourRightsInMonthsAndDay
       : parentalLeaveFormMessages.shared.yourRightsInMonthsAndDays
+
+  const rightsAfterChangeInDays = defaultMonths * daysInMonth - chosenGiveDays
+  const rightsAfterChangeInMonths = Math.floor(
+    (rightsAfterChangeInDays * 10000) / daysInMonth,
+  )
+  const rightsAfterChangeInWholeMonths =
+    Math.floor(rightsAfterChangeInMonths / 10000) * 10000
+
+  const remainingMonths = Math.floor(rightsAfterChangeInMonths / 10000)
+  const monthRemainder =
+    rightsAfterChangeInMonths - rightsAfterChangeInWholeMonths
+  const remainingDays = Math.floor(
+    monthRemainder / Math.floor((1 / daysInMonth) * 10000),
+  )
```

Comment:
Isn't this missing / 10000?

Also, I'm not sure why you need integer math for this. The monthRemainder should be fine with floats, and since the only input is an integer `chosenGiveDays` it should be fine to round `remainingDays` to the closest whole number (instead of floor).

Reply:
This was unnecessarily complicated. I've refactored it to use modulus.